src/Makefile.am

```diff
@@ -6,17 +6,18 @@ lib_LTLIBRARIES = libadf.la
 libadf_la_SOURCES = adf_dev.c \
                     adf_dev_flop.c \
                     adf_dev_hd.c \
-                    adf_disk.c \
                     adf_raw.c \
                     adf_bitm.c \
                     adf_dump.c \
                     adf_util.c \
                     adf_env.c \
                     adf_dir.c \
                     adf_file.c \
+                    adf_file_block.c \
                     adf_cache.c \
                     adf_link.c \
-                    adf_salv.c
+                    adf_salv.c \
+                    adf_vol.c
 
 include_HEADERS = adf_defs.h \
```

Comment:
It's a really excellent decision to prefix the names of structures, e.g. `struct Volume` to `struct AdfVolume`, because these struct names were very generic, and there are plenty of other programs or even other libraries which might have a `struct File`  or `struct Device`

With that in mind, ADFLib scatters its public API across 23 include files, two of which aren't prefixed "adf_" (defendian.h and now prefix.h), and they'll get installed in /usr/include

Firstly, the non-prefixed installed headers really ought to be prefixed if they're going to be centrally installed. Alternatively, install headers as  `#include <adflib/defendian.h>` e.g. `src/adflib/defendian.h` installs to `/usr/include/adflib/defendian.h`

Secondly, the headers could be improved by dividing into what are purely private, internal functions and structures of the library, versus what is public (and therefore part of the API that you promise to all library users).

It could even be the case that all public API functions (and structures and defines needed for them) are defined in libadf.h, and then that becomes the _only_ header file that needs to be installed. All other headers can include libadf.h but libadf.h should not include the other headers

Reply:
Headers currently go to `/usr/include/adf` (for Debian/Ubuntu). For others - I haven't checked... Probably default will be `/usr/local/include/adf`.

Yes, there is a lot to do here as well. Not everything can be done in this PR, though...

Reply:
That is not the case. The automake variable is `include_HEADERS` so whatever files are listed there go straight to `$(includedir)` which is  [`$(prefix)/include](https://www.gnu.org/software/automake/manual/html_node/Standard-Directory-Variables.html`)

```
$ make install
...
 /usr/bin/mkdir -p '/usr/local/include'
 /usr/bin/install -c -m 644 adf_defs.h adf_dev.h adf_dev_flop.h adf_dev_hd.h adf_blk.h adf_err.h adf_str.h adflib.h adf_bitm.h adf_cache.h adf_dir.h adf_dump.h adf_env.h adf_file.h adf_file_block.h adf_link.h adf_raw.h adf_salv.h adf_util.h adf_vol.h defendian.h hd_blk.h prefix.h '/usr/local/include'
...
```

If they _should_ go in $(prefix)/include/adf/ then automake should do that itself, so that all Linux distros can benefit from it (including Debian). It should not be wrongly installed in the base build but corrected in a Debian-specific script.

This is how you could make them install to $(prefix)/include/adf automatically:

```
adfincdir = $(includedir)/adf
adfinc_HEADERS = adf_defs.h adf_dev.h adf_dev_flop.h adf_dev_hd.h adf_blk.h adf_err.h adf_str.h adflib.h adf_bitm.h adf_cache.h adf_dir.h adf_dump.h adf_env.h adf_file.h adf_file_block.h adf_link.h adf_raw.h adf_salv.h adf_util.h adf_vol.h defendian.h hd_blk.h prefix.h
```

Reply:
As I wrote, I am checking/using mostly Debian builds and, somewhat, the others - but in the second case _without_ installation and/or using it for anything else (like building some app with it). So, as you suggest, this might indeed require improvements. However, I would insist that it would be done in compatibility with deb and with `pkg-config` (so we either update also deb configs or make changes in a non-breaking way).

Btw. I proposed putting headers in `[...]/include/adf` but of course this is not final. We can change it as well (eg. to `adflib` as you proposed above), but for sure they should go to a separated subdir, not to the main/system one.